internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -527,11 +538,15 @@ func (h *Host) ExecProcess(ctx context.Context, containerID string, params prot.
 		} else {
 			// Windows uses a different field for command, there's no enforcement
 			// around this yet for Windows so this is Linux specific at the moment.
-			err = h.securityPolicyEnforcer.EnforceExecInContainerPolicy(containerID, params.OCIProcess.Args, params.OCIProcess.Env, params.OCIProcess.Cwd)
+			envToKeep, err = h.securityPolicyEnforcer.EnforceExecInContainerPolicy(containerID, params.OCIProcess.Args, params.OCIProcess.Env, params.OCIProcess.Cwd)
 			if err != nil {
 				return pid, errors.Wrapf(err, "exec in container denied due to policy")
 			}
 
+			if envToKeep != nil {
+				params.OCIProcess.Args = envToKeep
```

Comment:
yeah, ok. we missed this...

Reply:
missed what?

Reply:
that we're assigning env to args...

Reply:
its how drop env needs to work at some level. currently the "protection" for this is in the engine so that the list can only contain items that were in the original list.

Reply:
what I'm saying is that instead of assinging to `OCIProcess.Env` we're assining to `OCIProcess.Args` in the code above. effectively overriding the process command rather than environment.

Reply:
ah yes, we need to get that fixed asap.